### ISSUE TITLE
`comparison_chain`: don't warn unless all branches are used

### ISSUE THREAD

**Issue by tgross35** [rust-lang/rust-clippy#14624]
### Summary

`comparison_chain` probably doesn't need to be raised unless all three branches are used, i.e. if there is a final `else` condition. It'

### Lint Name

comparison_chain

### Reproducer

Link: https://play.rust-lang.org/?version=nightly&mode=debug&edition=2024&gist=256e04cad4d4ef0d7cbdaeaf91309aad

I tried this code:

```rust
pub fn foo(a: i32) -> i32 {
    if a < 0 {
        return 5;
    } else if a == 0 {
        return 10;
    }

    unimplemented!();
}
```

I saw this happen:

```
warning: `if` chain can be rewritten with `match`
 --> src/lib.rs:4:5
  |
4 | /     if a < 0 {
5 | |         return 5;
6 | |     } else if a == 0 {
7 | |         return 10;
8 | |     }
  | |_____^ help: consider rewriting the `if` chain with `match`: `match a.cmp(&0) {...}`
  |
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#comparison_chain
note: the lint level is defined here
 --> src/lib.rs:1:9
  |
1 | #![warn(clippy::comparison_chain)]
  |    
```

Which would be

```rust
    match a.cmp(&0) {
        Ordering::Less => return 5,
        Ordering::Equal => return 10,
        Ordering::Greater => {}
    }
```

I expected to see this happen:

No output

### Version

```text
1.88.0-nightly
(2025-04-07 e643f59f6da3a84f43e7)
```

### Additional Labels

@rustbot label +L-suggestion

**Comment by tgross35** [comment]
Not a bug

@rustbot label -C-bug

**Comment by profetia** [comment]
I think this is what the lint intends. Not really a false positive.
